### PR TITLE
EES: 5278

### PR DESCRIPTION
**Pull request by saicharan2789**
This PR addresses the intermittent ui test failures of below test suites in pipeline -

1. `archive_publication.robot`
2.  `Data_catalogue. Robot`
3.  `Subject_reordering.robot`

### CODE REVIEW

File: tests/robot-tests/tests/libs/common.robot [hunk 1]
```diff
@@ -1014,3 +1013,20 @@ user takes html snapshot of element
 
 user waits for caches to expire
     sleep    %{WAIT_CACHE_EXPIRY}
+
+user wait for option to be available and select it
+    [Arguments]  ${dropdown_locator}  ${option_text}  ${timeout}=%{TIMEOUT}
+    wait until keyword succeeds  ${timeout}  1s  check option exist in dropdown  ${dropdown_locator}  ${option_text}
```
Comment (mmyoungman) [inline tests/robot-tests/tests/libs/common.robot:1019]: Not seen `wait until keyword succeeds` - looks like that could help us a lot with flaky tests :)

File: tests/robot-tests/tests/libs/common.robot [hunk 2]
```diff
@@ -1014,3 +1013,20 @@ user takes html snapshot of element
...
+check option exist in dropdown
+    [Arguments]  ${dropdown_locator}  ${option_text}
+    ${options}=  get webelements  ${dropdown_locator} > option
+    ${matched}=  set variable  False
+
+    FOR  ${option}  IN  @{options}
+    ${text}=  get text  ${option}
+    run keyword if  '${text}' == '${option_text}'  Set Variable  ${matched}  True
+    should be true  ${matched}  Option ${option_text} not found in the dropdown
+    END
```
Comment (mmyoungman) [inline tests/robot-tests/tests/libs/common.robot:1031]: I think this for loop isn't going to work. The final line in the for loop needs to be outside the loop, as right now this loops expects *every* element in `options` to match with `${option_text}`.

```
    FOR  ${option}  IN  @{options}
    ${text}=  get text  ${option}
    run keyword if  '${text}' == '${option_text}'  Set Variable  ${matched}  True
    END

    should be true  ${matched}  Option ${option_text} not found in the dropdown
```

_Hopefully_ that'll do it.

File: tests/robot-tests/tests/libs/common.robot (hunk 1 above)
Comment (saicharan2789) [reply tests/robot-tests/tests/libs/common.robot:1019]: yeah I have seen few people talking in the stack overflow discussing about wait until keyword succeeds keyword. Here is the one of the link - https://stackoverflow.com/questions/51501685/is-there-any-keyword-to-serve-as-wait-until-keywords-succeed

Comment (saicharan2789) [reply tests/robot-tests/tests/libs/common.robot:1019]: yeah it reduces flakiness a lot

File: tests/robot-tests/tests/libs/common.robot (hunk 2 above)
Comment (saicharan2789) [reply tests/robot-tests/tests/libs/common.robot]: I managed tried the way that you have mentioned, it looks like its doing linear search (checking each element against set of element). Due to which its failing. May be I am working on different approach, seems promising . Once its done, I will push latest changes

File: tests/robot-tests/tests/admin_and_public/bau/archive_publication.robot [hunk 3]
```diff
@@ -252,19 +253,23 @@ Check data catalogue page contains archive and superseding publication subjects
 
     user wait for option to be available and select it    css:select[id="filters-form-theme"]    %{TEST_THEME_NAME}
 
-    user checks select contains option    id:filters-form-publication    ${PUBLICATION_NAME_ARCHIVE}
-    user checks select contains option    id:filters-form-publication    ${PUBLICATION_NAME_SUPERSEDE}
+    user checks select contains option    css:select[id="filters-form-publication"]     ${PUBLICATION_NAME_ARCHIVE}
```
Comment (saicharan2789) [inline tests/robot-tests/tests/admin_and_public/bau/archive_publication.robot:256]: To make it consistent across the PR, I have updated the locator strategy with css selectors.

File: tests/robot-tests/tests/admin_and_public/bau/archive_publication.robot [hunk 4]
```diff
@@ -252,19 +253,23 @@ Check data catalogue page contains archive and superseding publication subjects
...
+    user checks select contains option    css:select[id="filters-form-publication"]     ${PUBLICATION_NAME_SUPERSEDE}
 
     user wait for option to be available and select it    css:select[id="filters-form-publication"]    ${PUBLICATION_NAME_SUPERSEDE}
+    sleep    1  # wait a moment to wait for release filter options to get updated
```
Comment (saicharan2789) [inline tests/robot-tests/tests/admin_and_public/bau/archive_publication.robot:260]: DOM is taking a bit longer to get updates. Due to which its requires a bit of 1 sec to get updated.

**Review by mmyoungman — APPROVED** [review]

**Pull request merged**